crates/sui/src/client_commands.rs

```diff
@@ -3021,3 +3177,59 @@ async fn execute_dry_run(
         .map_err(|e| anyhow!("Dry run failed: {e}"))?;
     Ok(SuiClientCommandResult::DryRun(response))
 }
+
+/// Call a dry run with the transaction data to estimate the gas budget.
+/// The estimated gas budget is computed as following:
+/// * the maximum between A and B, where:
+/// A = computation cost + GAS_SAFE_OVERHEAD * reference gas price
+/// B = computation cost + storage cost - storage rebate + GAS_SAFE_OVERHEAD * reference gas price
+/// overhead
+///
+/// This gas estimate is computed exactly as in the TypeScript SDK (sha 3c43692). Check the
+/// function `prepare` in the TransactionBlock.ts file and look for safeOverhead variable
+/// <https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/transactions/TransactionBlock.ts>
```

Comment:
nit: it would be nice to link to the exact sha you're referring to -- the current link is to `main`.

Reply:
Fixed